muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/hops/rest/HopsRestService.java

```diff
@@ -513,9 +526,14 @@ public Response listWorkspaceSuggestions(@QueryParam("subject") String subject,
             String dataSource = id[1];
             String identifier = id[0];
             SchoolDataIdentifier workspaceIdentifier = new SchoolDataIdentifier(identifier, dataSource);
+            
+            WorkspaceType workspaceType = null;
+            
             WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(workspaceIdentifier.getDataSource(), workspaceIdentifier.getIdentifier());
+            
             if (workspaceEntity != null) {
-              onlySignupWorkspaces = true;
+              Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
```

Comment:
Add a TODO here. We have to fetch workspace from Pyramus to fetch workspace type from Pyramus. Adding workspace type to Elastic would eliminate the need for both calls. This change is too big to implement in this PR but should be refactored later in order to improve performance.